get.sh

```diff
@@ -160,6 +160,34 @@ parseCommandLineArgs()
 	echo "TESTDIR: $TESTDIR"
 }
 
+# If the current directory contains only a single directory then squash that directory into the current directory
+squashSingleFolderContentsToCurrentDir()
+{
+        # Does current directory contain one and ONLY one directory?
+        if [[ `ls -d */ | wc -l` -eq 1 ]] && [[ `ls -d * | wc -l` -eq 1 ]]; then
+          folder=`ls -d */`
+          echo "Removing top-level folder ${folder}"
+          mv ${folder}* .
+          rmdir "${folder}"
+        fi
+}
+
+# Moves the given directory safely, ensuring the target does not exist, fail with error if it does exist
+moveDirectorySafely()
+{
+	if [[ -z ${1+x} ]] || [[ -z ${2+x} ]]; then
```

Comment:
Assuming this is just trying to trap the case where fewer than two parameters are provided, this seems less complex
```suggestion
	if [ $# -lt 2 ]; then
```